forcing_tools/netcdf2vic.py

```diff
@@ -0,0 +1,372 @@
+    writing to disk
+    '''
+
+    def __init__(self, lat='', lon='', x='', y='', prefix='', griddecimal=4,
+                 outpath='./'):
+        '''Defines x and y variables'''
+        self.lat = lat
+        self.lon = lon
+        self.x = x
+        self.y = y
+        self.filename = os.path.join(outpath,
+                                     "{0}_{1:.{prec}f}_"
+                                     "{2:.{prec}f}".format(prefix, lon, lat,
+                                                           prec=griddecimal))
+
+    def _open_binary(self):
+        print('opening binary file: {0}'.format(self.filename))
+        self.f = open(self.filename, 'wb')
+
+    def _open_ascii(self):
+        print('opening ascii file: {0}'.format(self.filename))
+        self.f = open(self.filename, 'w')
+
+    def _append_ascii(self, data):
+        np.savetxt(self.f, data, fmt=self.float_format)
+        return
+
+    def _append_binary(self, data):
+        for row in data:
+            d = struct.pack(self.binary_type, *row)
+            self.f.write(d)
+        return
+
+    def close(self):
+        print('closing file: {0}'.format(self.filename))
+        try:
+            self.f.close()
+        except:
+            pass
+
+    def __str__(self):
+        return "Point({0},{1},{2},{3})".format(self.lat, self.lon,
+                                               self.y, self.x)
+
+    def __repr__(self):
+        return "Point(lat={0}, lon={1}, \
+                      y={2}, x={3}, \
+                      filename={4})".format(self.lat,
+                                            self.lon,
+                                            self.y,
+                                            self.x,
+                                            self.filename)
+
+# -------------------------------------------------------------------- #
+
+
+# -------------------------------------------------------------------- #
+def main():
+    config_dict = process_command_line()
+
+    points, ncfiles = init(config_dict)
+
+    points = run(config_dict, ncfiles, points)
+
+    final(config_dict, ncfiles, points)
+# -------------------------------------------------------------------- #
+
+
+# -------------------------------------------------------------------- #
+def init(config_dict):
+
+    # ---------------------------------------------------------------- #
+    # Read domain file
+    domain_dict = config_dict['DOMAIN']
+    dom_data, dom_vatts, dom_gatts = read_domain(config_dict['DOMAIN'])
+    ys, xs = np.nonzero(dom_data[domain_dict['MASK_VAR']])
+    # ---------------------------------------------------------------- #
+
+    # ---------------------------------------------------------------- #
+    # Make sure OUTPATH is available
+    if not os.path.exists(config_dict['OPTIONS']['OUTPATH']):
+        os.makedirs(config_dict['OPTIONS']['OUTPATH'])
+    # ---------------------------------------------------------------- #
```

Comment:
I think it is better to use the `exist_ok=True` option here to remove the if statement